misc/dbt-materialize/dbt/include/materialize/macros/blue_green_deployment/create_deployment_environment.sql

```diff
@@ -0,0 +1,141 @@
+-- Copyright Materialize, Inc. and contributors. All rights reserved.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License in the LICENSE file at the
+-- root of this repository, or online at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+{% macro create_deployment_environment() %}
+
+{% set current_target_name = target.name %}
+{% set deployment = var('deployment') %}
+{% set target_config = deployment[current_target_name] %}
+
+-- Check if the target-specific configuration exists
+{% if not target_config %}
+    {{ exceptions.raise_compiler_error("No deployment configuration found for target " ~ current_target_name) }}
+{% endif %}
+
+{{ log("Creating deployment environment for target " ~ current_target_name, info=True) }}
+
+{% set clusters = target_config.get('clusters', []) %}
+{% set schemas = target_config.get('schemas', []) %}
+
+-- Check that all production schemas
+-- and clusters already exist
+{% for schema in schemas %}
+    {% if not schema_exists(schema.prod) %}
+        {{ exceptions.raise_compiler_error("Production schema " ~ schema.prod ~ " does not exist") }}
+    {% endif %}
+{% endfor %}
+
+{% for cluster in clusters %}
+    {% if not cluster_exists(cluster.prod) %}
+        {{ exceptions.raise_compiler_error("Production cluster " ~ cluster.prod ~ " does not exist") }}
+    {% endif %}
+{% endfor %}
+
+{% for schema in schemas %}
+    {% if schema_exists(schema.prod_deploy) %}
+        {{ log("Deployment schema " ~ schema.prod_deploy ~ " already exists", info=True)}}
+        {% set schema_empty %}
+            SELECT *
+            FROM mz_objects
+           JOIN mz_schemas ON mz_objects.schema_id = mz_schemas.id
+            WHERE mz_schemas.name = lower(trim('{{ schema.prod_deploy }}'))
+                AND mz_objects.id LIKE 'u%'
+        {% endset %}
+
+        {% if run_query(schema_empty)|length > 0 %}
+            {{ log("[Warning] Deployment schema " ~ schema.prod_deploy ~ " is not empty", info=True) }}
+            {{ log("[Warning] Confirm the objects it contains are expected before deployment", info=True) }}
```

Comment:
Open design question: right now if the deployment schema or cluster already exists and contains user objects, we issue a warning. Should this instead be an error by default with a flag to downgrade it to a warning?

Reply:
Well, tried something and ended up breaking the tests. I’ll fix it tomorrow